test/src/integration/onChainBlock.test.ts

```diff
@@ -0,0 +1,408 @@
+
+        const sdk = node.sdk;
+
+        await sdk.rpc.devel.startSealing();
+        await sdk.rpc.devel.startSealing();
+
+        const genesisBlock = await sdk.rpc.chain.getBlock(0);
+        const block1 = await sdk.rpc.chain.getBlock(1);
+        const block2 = await sdk.rpc.chain.getBlock(2);
+
+        await node.clean();
+        soloGenesisBlock = new Header(
+            genesisBlock.parentHash,
+            new U256(genesisBlock.timestamp),
+            new U256(genesisBlock.number),
+            genesisBlock.author.accountId,
+            Buffer.from(genesisBlock.extraData),
+            genesisBlock.parcelsRoot,
+            genesisBlock.stateRoot,
+            genesisBlock.invoicesRoot,
+            genesisBlock.score,
+            genesisBlock.seal
+        );
+        soloBlock1 = new Header(
+            soloGenesisBlock.hashing(),
+            new U256(block1.timestamp),
+            new U256(block1.number),
+            block1.author.accountId,
+            Buffer.from(block1.extraData),
+            block1.parcelsRoot,
+            block1.stateRoot,
+            block1.invoicesRoot,
+            new U256(2222222222222),
+            block1.seal
+        );
+        soloBlock2 = new Header(
+            soloBlock1.hashing(),
+            new U256(block2.timestamp),
+            new U256(block2.number),
+            block2.author.accountId,
+            Buffer.from(block2.extraData),
+            block2.parcelsRoot,
+            block2.stateRoot,
+            block2.invoicesRoot,
+            new U256(33333333333333),
+            block2.seal
+        );
+    });
+
+    beforeEach(async () => {
+        nodeA = new CodeChain({ logFlag: true });
+        await nodeA.start();
+    });
+
+    afterEach(async () => {
+        await nodeA.clean();
+    });
+
+    test(
+        "OnChain invalid timestamp block propagation test",
+        async () => {
+            const TH = new TestHelper("0.0.0.0", nodeA.port);
+            await TH.establish();
+
+            // TH.setLog();
+            const sdk = nodeA.sdk;
+
+            // Genesis block
+            const header = soloGenesisBlock;
+
+            // Block 1
+            const header1 = soloBlock1;
+
+            // Block 2
+            const header2 = soloBlock2;
+
+            await TH.sendEncodedBlock(
+                [
+                    header.toEncodeObject(),
+                    header1.toEncodeObject(),
+                    header2.toEncodeObject()
+                ],
+                [[], []],
+                header2.hashing(),
+                header2.getScore()
+            );
+
+            const block1 = await sdk.rpc.chain.getBlock(1);
+            const block2 = await sdk.rpc.chain.getBlock(2);
+
+            expect(block1).toEqual(expect.anything());
+            expect(block2).toEqual(expect.anything());
+
+            await TH.end();
+        },
+        30000
+    );
+
+    test(
+        "OnChain invalid timestamp block propagation test",
```

Comment:
I think it should tell the difference between the previous one. They have the same description.